cmd/dlt/admin/cmd.go

```diff
@@ -45,85 +45,93 @@ func run(cmd *cobra.Command, _ []string) {
+		r.Reporter.Errorf("Failed to get HTPasswd identity provider for cluster '%s': %v", r.ClusterKey, err)
 		os.Exit(1)
 	}
 
-	var htpasswdIDP *cmv1.IdentityProvider
+	var identityProvider *cmv1.IdentityProvider
 	for _, item := range idps {
 		if ocm.IdentityProviderType(item) == ocm.HTPasswdIDPType {
-			htpasswdIDP = item
+			identityProvider = item
 		}
 	}
-	if htpasswdIDP == nil {
-		r.Reporter.Errorf("Cluster '%s' does not have an admin user", clusterKey)
+	if identityProvider == nil {
+		r.Reporter.Errorf("Cluster '%s' does not have an admin user", r.ClusterKey)
 		os.Exit(1)
 	}
 
-	if confirm.Confirm("delete %s user on cluster %s", idp.ClusterAdminUsername, clusterKey) {
+	if confirm.Confirm("delete %s user on cluster %s", idp.ClusterAdminUsername, r.ClusterKey) {
 		// delete `cluster-admin` user from the HTPasswd IDP
-		r.Reporter.Debugf("Deleting user '%s' identity provider on cluster '%s'", idp.ClusterAdminUsername, clusterKey)
-		err = r.OCMClient.DeleteUser(cluster.ID(), "cluster-admins", idp.ClusterAdminUsername)
+		r.Reporter.Debugf("Deleting user '%s' from cluster-admins group on cluster '%s'",
+			idp.ClusterAdminUsername, r.ClusterKey)
+		err = r.OCMClient.DeleteUser(clusterID, "cluster-admins", idp.ClusterAdminUsername)
 		if err != nil {
 			r.Reporter.Errorf("Failed to delete '%s' user from cluster-admins groups of cluster '%s': %s",
-				idp.ClusterAdminUsername, clusterKey, err)
-			os.Exit(1)
-		}
-
-		//delete now the cluster-admin user from the htpasswd idp
-		users, err := r.OCMClient.GetHTPasswdUserList(cluster.ID(), htpasswdIDP.ID())
-		if err != nil {
-			r.Reporter.Errorf("Failed to list htpasswd idp users of cluster '%s': %s",
-				clusterKey, err)
-			os.Exit(1)
-		}
-		for _, user := range users.Slice() {
-			if user.Username() == idp.ClusterAdminUsername {
-				err := r.OCMClient.DeleteHTPasswdUser(idp.ClusterAdminUsername, cluster.ID(), htpasswdIDP)
-				if err != nil {
-					r.Reporter.Errorf("Failed to delete '%s' user from htpasswd idp users list of cluster '%s': %s",
-						idp.ClusterAdminUsername, clusterKey, err)
-					os.Exit(1)
-				}
-
-			}
-		}
-
-		users, err = r.OCMClient.GetHTPasswdUserList(cluster.ID(), htpasswdIDP.ID())
-		if err != nil {
-			r.Reporter.Errorf("Failed to list htpasswd idp users of cluster '%s': %s",
-				clusterKey, err)
+				idp.ClusterAdminUsername, r.ClusterKey, err)
 			os.Exit(1)
 		}
 
-		htpasswdIdentityProvider, ok := htpasswdIDP.GetHtpasswd()
+		htpasswdIdp, ok := identityProvider.GetHtpasswd()
 		if !ok {
-			r.Reporter.Errorf("Failed to get htpasswd idp of cluster '%s': %s",
-				clusterKey, err)
+			r.Reporter.Errorf("Failed to get htpasswd idp for cluster '%s'", clusterID)
 			os.Exit(1)
 		}
-
-		if users.Len() == 0 && htpasswdIdentityProvider.Username() == "" {
-			//delete the idp as users list is empty
-			err := r.OCMClient.DeleteIdentityProvider(cluster.ID(), htpasswdIDP.ID())
+		if htpasswdIdp.Username() == "cluster-admin" {
```

Comment:
Not liking this if else, will take a look at a strategy pattern in golang to set this up

Reply:
That logic L85-90 better be encapsulated inside a func `wasAdminCreatedUsingOldRosa()`
That condition inside it might change, but shouldn't impact how we handle it